bike-erp/backend/src/main/routes/component_routes.ts

```diff
@@ -42,7 +42,19 @@ router.post("/addComponent", authenticateToken, verifyRole([Role.ADMIN, Role.MAN
   })
 })
 
-router.get("/:component_id", authenticateToken, verifyRole([Role.ADMIN, Role.MANAGER, Role.EMPLOYEE, Role.CUSTOMER]), (req, res) => {
+router.get("/componentByLocation", authenticateToken, (req, res) => {
+  const loc = req.query.location;
+  inventoryManagementService
+    .getComponentsByLocation(loc)
+    .then((response) => {
+      res.json(response);
+    })
+    .catch((error) => {
+      res.status(error.status).send(error.message);
+    });
+});
+
+router.get("/:component_id", authenticateToken, verifyRole([Role.ADMIN, Role.MANAGER, Role.EMPLOYEE]), (req, res) => {
```

Comment:
You need to add the following here:
![image](https://user-images.githubusercontent.com/48056716/115464208-b8d82100-a1fa-11eb-89c1-959de3c818df.png)
When you try to order a bike as a customer you get unauthorized user:
![image](https://user-images.githubusercontent.com/48056716/115464247-cbeaf100-a1fa-11eb-9920-4665d65a444f.png)

Reply:
router.get("/:component_id", authenticateToken, verifyRole([Role.ADMIN, Role.MANAGER, Role.EMPLOYEE, Role.CUSTOMER]), (req, res) => {

Reply:
good catch. don't know how that slipped it should be fixed